front/app/components/ScreenReadableEventDate/index.tsx

```diff
@@ -0,0 +1,47 @@
+import React from 'react';
+
+import moment from 'moment';
+
+import { IEventData } from 'api/events/types';
+
+import { ScreenReaderOnly } from 'utils/a11y';
+import { useIntl } from 'utils/cl-intl';
+
+import messages from './messages';
+
+interface Props {
+  event: IEventData;
+}
+
+const ScreenReadableEventDate = ({ event }: Props) => {
+  const { formatMessage } = useIntl();
+  const startAtMoment = moment(event.attributes.start_at);
+  const endAtMoment = moment(event.attributes.end_at);
+  const isEventMultipleDays =
+    startAtMoment.dayOfYear() !== endAtMoment.dayOfYear();
+
+  return (
+    <ScreenReaderOnly>
+      {isEventMultipleDays ? (
+        <p>
+          {formatMessage(messages.multiDayScreenReaderDate, {
+            startDate: startAtMoment.format('MMMM Do, YYYY'),
+            startTime: startAtMoment.format('LT'),
+            endDate: endAtMoment.format('MMMM Do, YYYY'),
+            endTime: endAtMoment.format('LT'),
+          })}
+        </p>
+      ) : (
+        <p>
+          {formatMessage(messages.singleDayScreenReaderDate, {
+            eventDate: startAtMoment.format('MMMM Do, YYYY'),
+            startTime: startAtMoment.format('LT'),
+            endTime: endAtMoment.format('LT'),
+          })}
+        </p>
+      )}
+    </ScreenReaderOnly>
```

Comment:
```suggestion
    <ScreenReaderOnly>
      {formatMessage(
        isEventMultipleDays
          ? messages.multiDayScreenReaderDate
          : messages.singleDayScreenReaderDate,
        {
          startDate: startAtMoment.format('MMMM Do, YYYY'),
          startTime: startAtMoment.format('LT'),
          endDate: endAtMoment.format('MMMM Do, YYYY'),
          endTime: endAtMoment.format('LT'),
        }
      )}
    </ScreenReaderOnly>
```
The only change due to the condition is the message, so you can also do the above for less code. I also removed `p`. Unless you added that for a specific reason, it's not necessary.